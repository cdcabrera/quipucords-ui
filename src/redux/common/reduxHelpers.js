import { helpers } from '../../common/helpers';

const FULFILLED_ACTION = base => `${base}_FULFILLED`;

const PENDING_ACTION = base => `${base}_PENDING`;

const REJECTED_ACTION = base => `${base}_REJECTED`;

const setStateProp = (prop, data, options) => {
  const { state = {}, initialState = {}, reset = true } = options;
  let obj = { ...state };

  if (process.env.REACT_APP_ENV === 'development' && prop && !state[prop]) {
    console.error(`Error: Property ${prop} does not exist within the passed state.`, state);
  }

  if (process.env.REACT_APP_ENV === 'development' && reset && prop && !initialState[prop]) {
    console.warn(`Warning: Property ${prop} does not exist within the passed initialState.`, initialState);
  }

  if (reset && prop) {
    obj[prop] = {
      ...state[prop],
      ...initialState[prop],
      ...data
    };
  } else if (reset && !prop) {
    obj = {
      ...state,
      ...initialState,
      ...data
    };
  } else if (prop) {
    obj[prop] = {
      ...state[prop],
      ...data
    };
  } else {
    obj = {
      ...state,
      ...data
    };
  }

  return obj;
};

const generatedPromiseActionReducer = (types = [], state = {}, action = {}) => {
  if (!action.meta || !action.meta.id) {
    return state;
  }

  const { type } = action;
  const expandedTypes = [];

  types.forEach(
    val =>
      (Array.isArray(val.type) && val.type.forEach(subVal => expandedTypes.push({ ref: val.ref, type: subVal }))) ||
      expandedTypes.push(val)
  );

  const [whichType] = expandedTypes.filter(val =>
    new RegExp(
      `^(${REJECTED_ACTION(val.type || val)}|${PENDING_ACTION(val.type || val)}|${FULFILLED_ACTION(val.type || val)})$`
    ).test(type)
  );

  if (!whichType) {
    return state;
  }

  switch (type) {
    case REJECTED_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        {
          [action.meta.id]: {
            error: action.error,
            errorMessage: helpers.getMessageFromResults(action.payload).message,
            errorStatus: helpers.getStatusFromResults(action.payload),
            metaData: action.meta.data,
            metaId: action.meta.id,
            metaQuery: action.meta.query
          }
        },
        {
          state
        }
      );
    case PENDING_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        {
          [action.meta.id]: {
            metaData: action.meta.data,
            metaId: action.meta.id,
            metaQuery: action.meta.query,
            pending: true
          }
        },
        {
          state
        }
      );

    case FULFILLED_ACTION(whichType.type || whichType):
      return setStateProp(
        whichType.ref || null,
        {
          [action.meta.id]: {
            data: (action.payload && action.payload.data) || {},
            fulfilled: true,
            metaData: action.meta.data,
            metaId: action.meta.id,
            metaQuery: action.meta.query
          }
        },
        {
          state
        }
      );

    default:
      return state;
  }
};

const reduxHelpers = {
  FULFILLED_ACTION,
  PENDING_ACTION,
  REJECTED_ACTION,
  generatedPromiseActionReducer,
  setStateProp
};

export { reduxHelpers as default, reduxHelpers };
