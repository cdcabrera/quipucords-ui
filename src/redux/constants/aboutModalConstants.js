const ABOUT_MODAL_SHOW = 'ABOUT_MODAL_SHOW';
const ABOUT_MODAL_HIDE = 'ABOUT_MODAL_HIDE';

const aboutModalTypes = {
  ABOUT_MODAL_SHOW,
  ABOUT_MODAL_HIDE
};

export { aboutModalTypes as default, aboutModalTypes, ABOUT_MODAL_SHOW, ABOUT_MODAL_HIDE };
