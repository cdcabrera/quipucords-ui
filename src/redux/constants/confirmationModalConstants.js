const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

const confirmationModalTypes = {
  CONFIRMATION_MODAL_SHOW,
  CONFIRMATION_MODAL_HIDE
};

export { confirmationModalTypes as default, confirmationModalTypes, CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
