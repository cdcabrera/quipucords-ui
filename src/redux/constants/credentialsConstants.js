const ADD_CREDENTIAL = 'ADD_CREDENTIAL';
const DELETE_CREDENTIAL = 'DELETE_CREDENTIAL';
const GET_CREDENTIALS = 'GET_CREDENTIALS';
const UPDATE_CREDENTIAL = 'UPDATE_CREDENTIAL';
const CREATE_CREDENTIAL_SHOW = 'CREATE_CREDENTIAL_SHOW';
const EDIT_CREDENTIAL_SHOW = 'EDIT_CREDENTIAL_SHOW';
const UPDATE_CREDENTIAL_HIDE = 'UPDATE_CREDENTIAL_HIDE';

const SELECT_CREDENTIAL = 'SELECT_CREDENTIAL';
const DESELECT_CREDENTIAL = 'DESELECT_CREDENTIAL';
const EXPANDED_CREDENTIAL = 'EXPANDED_CREDENTIAL';
const NOT_EXPANDED_CREDENTIAL = 'NOT_EXPANDED_CREDENTIAL';
const UPDATE_CREDENTIALS = 'UPDATE_CREDENTIALS';

const credentialsTypes = {
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};

export {
  credentialsTypes as default,
  credentialsTypes,
  ADD_CREDENTIAL,
  UPDATE_CREDENTIAL,
  DELETE_CREDENTIAL,
  GET_CREDENTIALS,
  CREATE_CREDENTIAL_SHOW,
  EDIT_CREDENTIAL_SHOW,
  UPDATE_CREDENTIAL_HIDE,
  SELECT_CREDENTIAL,
  DESELECT_CREDENTIAL,
  EXPANDED_CREDENTIAL,
  NOT_EXPANDED_CREDENTIAL,
  UPDATE_CREDENTIALS
};
