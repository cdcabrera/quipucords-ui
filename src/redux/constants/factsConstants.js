const ADD_FACTS = 'ADD_FACTS';

const factsTypes = {
  ADD_FACTS
};

export { factsTypes as default, factsTypes, ADD_FACTS };
