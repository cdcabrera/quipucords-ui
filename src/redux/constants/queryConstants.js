import { apiTypes } from '../../constants/apiConstants';

/**
 * Inventory query types associated with only HOSTS' and INSTANCES' queries.
 */
const SET_QUERY_CREDENTIALS_TYPES = {
  [apiTypes.API_.DIRECTION]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DIRECTION}`,
  [RHSM_API_QUERY_TYPES.DISPLAY_NAME]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.DISPLAY_NAME}`,
  [RHSM_API_QUERY_TYPES.SORT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.SORT}`,
  [RHSM_API_QUERY_TYPES.LIMIT]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.LIMIT}`,
  [RHSM_API_QUERY_TYPES.OFFSET]: `SET_QUERY_RHSM_HOSTS_INVENTORY_${RHSM_API_QUERY_TYPES.OFFSET}`
};
