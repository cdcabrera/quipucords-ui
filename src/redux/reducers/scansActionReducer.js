import reduxHelpers from '../../common/reduxHelpers';
import { scansTypes } from '../constants';

const initialState = {
  connection: {},
  inspection: {},
  job: {},
  jobs: {},
  cancel: {},
  pause: {},
  restart: {},
  start: {}
};

const scansActionReducer = (state = initialState, action) =>
  reduxHelpers.generatedPromiseActionReducer(
    [
      { ref: 'connection', type: scansTypes.GET_SCAN_CONNECTION_RESULTS },
      { ref: 'inspection', type: scansTypes.GET_SCAN_INSPECTION_RESULTS },
      { ref: 'job', type: scansTypes.GET_SCAN_JOB },
      { ref: 'jobs', type: scansTypes.GET_SCAN_JOBS },
      { ref: 'cancel', type: scansTypes.CANCEL_SCAN },
      { ref: 'pause', type: scansTypes.PAUSE_SCAN },
      { ref: 'restart', type: scansTypes.RESTART_SCAN },
      { ref: 'start', type: scansTypes.START_SCAN }
    ],
    state,
    action
  );

scansActionReducer.initialState = initialState;

export { scansActionReducer as default, initialState, scansActionReducer };
