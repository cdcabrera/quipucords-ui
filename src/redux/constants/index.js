import { aboutModalTypes } from './aboutModalConstants';
import { confirmationModalTypes } from './confirmationModalConstants';
import { credentialsTypes } from './credentialsConstants';
import { factsTypes } from './factsConstants';
import { reportsTypes } from './reportsConstants';
import { scansTypes } from './scansConstants';
import { sourcesTypes } from './sourcesConstants';
import { statusTypes } from './statusConstants';
import { toastNotificationTypes } from './toasNotificationConstants';
import { userTypes } from './userConstants';
import { viewTypes } from './viewConstants';
import { viewPaginationTypes } from './viewPaginationConstants';
import { viewToolbarTypes } from './viewToolbarConstants';

const reduxTypes = {
  aboutModal: aboutModalTypes,
  confirmationModal: confirmationModalTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  status: statusTypes,
  toastNotifications: toastNotificationTypes,
  user: userTypes,
  view: viewTypes,
  viewPagination: viewPaginationTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes as default,
  reduxTypes,
  aboutModalTypes,
  confirmationModalTypes,
  credentialsTypes,
  factsTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  statusTypes,
  toastNotificationTypes,
  userTypes,
  viewTypes,
  viewPaginationTypes,
  viewToolbarTypes
};
