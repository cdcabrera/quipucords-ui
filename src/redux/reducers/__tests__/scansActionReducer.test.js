import { scansActionReducer } from '../scansActionReducer';
import { scansTypes as types } from '../../constants';
import { reduxHelpers } from '../../common/reduxHelpers';

describe('ScansActionReducer', () => {
  it('should return the initial state', () => {
    expect(scansActionReducer.initialState).toBeDefined();
  });

  it('should handle all defined error types', () => {
    const specificTypes = [
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        meta: {
          id: 1,
          data: {},
          query: {}
        },
        type: reduxHelpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = scansActionReducer(undefined, dispatched);

      expect({ type: reduxHelpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot(
        `rejected types ${value}`
      );
    });
  });

  it('should handle all defined pending types', () => {
    const specificTypes = [
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        meta: {
          id: 1,
          data: {},
          query: {}
        },
        type: reduxHelpers.PENDING_ACTION(value)
      };

      const resultState = scansActionReducer(undefined, dispatched);

      expect({ type: reduxHelpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot(
        `pending types ${value}`
      );
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [
      types.GET_SCAN_CONNECTION_RESULTS,
      types.GET_SCAN_INSPECTION_RESULTS,
      types.GET_SCAN_JOB,
      types.GET_SCAN_JOBS,
      types.CANCEL_SCAN,
      types.PAUSE_SCAN,
      types.RESTART_SCAN,
      types.START_SCAN
    ];

    specificTypes.forEach(value => {
      const dispatched = {
        meta: {
          id: 1,
          data: {},
          query: {}
        },
        type: reduxHelpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = scansActionReducer(undefined, dispatched);

      expect({ type: reduxHelpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
