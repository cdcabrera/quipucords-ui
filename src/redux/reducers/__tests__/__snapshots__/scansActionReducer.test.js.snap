// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScansActionReducer should handle all defined error types: rejected types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "CANCEL_SCAN_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOB_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOBS_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "restart": Object {},
    "start": Object {},
  },
  "type": "PAUSE_SCAN_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "start": Object {},
  },
  "type": "RESTART_SCAN_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined error types: rejected types START_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {
      "1": Object {
        "error": true,
        "errorMessage": Object {
          "message": "ERROR",
          "messages": Object {
            "detail": "ERROR",
          },
          "status": 0,
          "url": null,
        },
        "errorStatus": 0,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
  },
  "type": "START_SCAN_REJECTED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "CANCEL_SCAN_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOB_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOBS_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "restart": Object {},
    "start": Object {},
  },
  "type": "PAUSE_SCAN_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
    "start": Object {},
  },
  "type": "RESTART_SCAN_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined fulfilled types: fulfilled types START_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {
      "1": Object {
        "data": Object {
          "test": "success",
        },
        "fulfilled": true,
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
      },
    },
  },
  "type": "START_SCAN_FULFILLED",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types CANCEL_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "CANCEL_SCAN_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types GET_SCAN_CONNECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_CONNECTION_RESULTS_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types GET_SCAN_INSPECTION_RESULTS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_INSPECTION_RESULTS_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types GET_SCAN_JOB 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOB_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types GET_SCAN_JOBS 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "pause": Object {},
    "restart": Object {},
    "start": Object {},
  },
  "type": "GET_SCAN_JOBS_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types PAUSE_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "restart": Object {},
    "start": Object {},
  },
  "type": "PAUSE_SCAN_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types RESTART_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
    "start": Object {},
  },
  "type": "RESTART_SCAN_PENDING",
}
`;

exports[`ScansActionReducer should handle all defined pending types: pending types START_SCAN 1`] = `
Object {
  "result": Object {
    "cancel": Object {},
    "connection": Object {},
    "inspection": Object {},
    "job": Object {},
    "jobs": Object {},
    "pause": Object {},
    "restart": Object {},
    "start": Object {
      "1": Object {
        "metaData": Object {},
        "metaId": 1,
        "metaQuery": Object {},
        "pending": true,
      },
    },
  },
  "type": "START_SCAN_PENDING",
}
`;
